pyspectral/atm_correction_ir.py

```diff
@@ -128,12 +129,29 @@ def delta(z):
         DELTA_REF = 6.2
         return (1 + DELTA_REF)**ratio(z, Z_0, Z_REF) - 1
 
-    y0, x0 = np.ma.where(view_zen == 0)
-    data[y0, x0] += tau0(data[y0, x0])
+    if hasattr(data, 'compute') or hasattr(view_zen, 'compute'):
+        is_dask_data = True
+    else:
+        is_dask_data = False
 
-    y, x = np.ma.where((view_zen > 0) & (view_zen < 90) & (~data.mask))
-    data[y, x] += tau(data[y, x]) * delta(view_zen[y, x])
-    return data
+    if is_dask_data:
+        dask_data = da.where(view_zen == 0,
+                             data + tau0(data),
+                             da.where((view_zen > 0) & (view_zen < 90),
+                                      data + (tau(data) * delta(view_zen)),
+                                      data))
+        return dask_data
+    else:
+        y0, x0 = np.ma.where(view_zen == 0)
+        data[y0, x0] += tau0(data[y0, x0])
+
+        y, x = np.ma.where((view_zen > 0) & (view_zen < 90) & (~data.mask))
+        data[y, x] += tau(data[y, x]) * delta(view_zen[y, x])
+
+        LOG.debug('Calculated data min/avr/max: {:.3f}/{:.3f}/{:.3f}'.format(np.nanmin(data),
+                                                                             np.nanmean(data),
+                                                                             np.nanmax(data)))
```

Comment:
Was this log message here before?

Reply:
No, it wasn't. It's for debugging only (that's why it's has debug level :-) ) and only for numpy to avoid unnecessary computations for dask data. I can remove it if you want.

Reply:
Yeah, removing it would be good. It requires numpy to iterate over the data three times just for a debug message which is not good.